src/Container.ts

```diff
@@ -40,38 +49,42 @@ export class Container {
     /**
      * Registers a new service.
      *
-     * @param name Service name. Optional
-     * @param type Service class
-     * @param params Parameters to be sent to constructor on service initialization
+     * @param descriptor
      */
-    static registerService(name: string, type: Function, params?: any[]) {
-        this.registeredServices.push({ name: name, type: type, params: params });
+    static registerService(descriptor: ServiceDescriptor) {
+        this.registeredServices.push(descriptor);
     }
 
     /**
      * Retrieves the service with the specific name or given type from the service container.
-     * Optionally parameters can be pass in the case if instance is initialized in the container for the first time.
+     * Optionally, parameters can be passed in case if instance is initialized in the container for the first time.
      */
     static get<T>(type: ConstructorFunction<T>, params?: any[]): T;
     static get<T>(name: string, params?: any[]): T;
     static get<T>(typeOrName: ConstructorFunction<T>|string, params?: any[]): T {
 
         // normalize parameters
-        let type: Function, name: string;
+        let type: Function;
+        let name: string;
+        let factory: Function;
         if (typeof typeOrName === "string") {
             name = <string> typeOrName;
         } else {
             type = <Function> typeOrName;
         }
 
         // find if service was already registered
-        const registeredService = this.findRegisteredService(name, type);
-        // console.log("registeredService: ", registeredService);
-        if (registeredService) {
-            if (!type)
-                type = registeredService.type;
-            if (!params)
-                params = registeredService.params;
+        const serviceDescriptor = this.findRegisteredService(name, type);
+        if (serviceDescriptor) {
+            if (!type) {
+                type = serviceDescriptor.type;
+            }
+            if (!params) {
+                params = serviceDescriptor.params;
+            }
+            if (serviceDescriptor.factory) {
+                factory = serviceDescriptor.factory;
+            }
```

Comment:
there is no need to make such things standartized because sometimes its really better to put and sometimes its not. Better to choose best cases instead of following dummy (in this case) standartization. 

No need to show example where you can easily change "change line" - you can do it almost as well easily as putting {} and add extra line. Yeah I know it may be annoying sometimes but code beauty worth it, and its totally not a "lot of extraneous work", just few seconds change. 

I also changed my opinion multiple times during my career regarding to those braces. First without {} then with. Your arguments are standard PHP-guy arguments (I saw lot of them arguing this :D), I also used these arguments in past, but nowadays and already 4-5 years stable I prefer put them depend on case where necessary. This makes the code best and clear. Yeah I know code style disputes never stop